packages/experimental/Divider/src/Divider.styling.ts

```diff
@@ -0,0 +1,149 @@
+import { useMemo } from 'react';
+import { ViewProps } from 'react-native';
+import { Theme } from '@fluentui-react-native/framework';
+import { IconPropsV1 as IconProps } from '@fluentui-react-native/icon';
+import { TextProps } from '@fluentui-react-native/text';
+import { DividerTokens, DividerAppearance } from './Divider.types';
+
+const getIconProps = (tokens: DividerTokens, icon: IconProps): IconProps => {
+  if (icon.fontSource) {
+    return {
+      fontSource: {
+        ...icon.fontSource,
+        color: tokens.color || tokens.contentColor,
+      },
+    };
+  } else if (icon.svgSource) {
+    return {
+      svgSource: {
+        ...icon.svgSource,
+        color: tokens.color || tokens.contentColor,
+      },
+    };
+  } else {
+    throw new Error('IconProps require either a fontSource or svgSource; neither has been passed.');
+  }
+};
+
+export const useDividerSlotProps = (tokens: DividerTokens, icon?: IconProps) => {
```

Comment:
It feels like we're recreating the `useStyling` hook that `compose` uses here. I wonder if that hook is modular enough that we could actually use it with compressible? Might be worth looking into.